src/node/kernel_notifications.cpp

```diff
@@ -99,6 +100,14 @@ void KernelNotifications::fatalError(const bilingual_str& message)
                     m_exit_status, message, &m_warnings);
 }
 
+std::optional<uint256> KernelNotifications::TipBlock()
+{
+    AssertLockHeld(m_tip_block_mutex);
+    Assume(!m_tip_block || m_tip_block.value() != uint256{});
```

Comment:
nit: I tend to agree with other commenters that explicitly checking for non-zero and the comment about it might not be needed but I am not too passionate about it. Similarly I would add to that that this line here could have used `uint256::ZERO` like the code that was changed previously.

Reply:
There was another pull request where someone suggested the use of `uint256{}` instead of `uint256::ZERO`, though I forgot where. The codebase uses both. Though in this context perhaps `ZERO` makes the intention more clear.